Userland/Applications/PixelPaint/PaletteWidget.cpp

```diff
@@ -162,4 +136,87 @@ void PaletteWidget::set_secondary_color(Color color)
     m_secondary_color_widget->update();
 }
 
+Result<void, String> PaletteWidget::load_palette_file(String const& file_path)
+{
+    auto file_or_error = Core::File::open(file_path, Core::OpenMode::ReadOnly);
+    if (file_or_error.is_error())
+        return file_or_error.error();
+
+    auto& file = *file_or_error.value();
+
+    Vector<Color> new_colors;
+
+    while (file.can_read_line()) {
+        auto line = file.read_line();
+        auto color = Color::from_string(line);
+        if (!color.has_value()) {
+            return String::formatted("Could not parse \"{}\" as a color", line);
+        }
+
+        new_colors.append(color.value());
+    }
+
+    file.close();
+
+    populate_color_container(new_colors);
+
+    return {};
+}
+
+Result<void, String> PaletteWidget::save_palette_file(String const& file_path)
+{
+    auto file_or_error = Core::File::open(file_path, Core::OpenMode::WriteOnly);
+    if (file_or_error.is_error())
+        return file_or_error.error();
+
+    auto& file = *file_or_error.value();
+
+    for (auto& color_container : m_color_container->child_widgets()) {
+        color_container.for_each_child_of_type<ColorWidget>([&](auto& color_widget) {
+            file.write(color_widget.color().to_string_without_alpha());
+            file.write("\n");
+            return IterationDecision::Continue;
+        });
+    }
+
+    file.close();
+
+    return {};
+}
+
+void PaletteWidget::populate_color_container(Vector<Color> const& colors)
+{
+    int colors_to_add = colors.size();
+    if (colors_to_add == 0) {
+        dbgln("No colors to populate the container with :(");
+        return;
+    }
+
+    auto& top_color_container = *m_color_container->find_descendant_of_type_named<GUI::Widget>("top_color_container");
+    top_color_container.remove_all_children();
+
+    auto& bottom_color_container = *m_color_container->find_descendant_of_type_named<GUI::Widget>("bottom_color_container");
+    bottom_color_container.remove_all_children();
+
+    auto add_color_widget = [&](GUI::Widget& container, Color color) {
+        auto& color_widget = container.add<ColorWidget>(color, *this);
+        color_widget.set_fill_with_background_color(true);
+        auto pal = color_widget.palette();
+        pal.set_color(ColorRole::Background, color);
+        color_widget.set_palette(pal);
+    };
+
+    int colors_per_row = ceil(colors_to_add / 2);
+    int number_of_added_colors = 0;
+    for (auto& color : colors) {
+        if (number_of_added_colors < colors_per_row) {
+            add_color_widget(top_color_container, color);
+        } else {
+            add_color_widget(bottom_color_container, color);
+        }
```

Comment:
No need for brackets when there's a single line of code:
```suggestion
        if (number_of_added_colors < colors_per_row)
            add_color_widget(top_color_container, color);
        else
            add_color_widget(bottom_color_container, color);
```